internal/telemetry/prometheus.go

```diff
@@ -16,50 +16,58 @@ package telemetry
 
 // Taken from https://opencensus.io/quickstart/go/metrics/#1
 import (
-	"net/http"
-
 	ocPrometheus "contrib.go.opencensus.io/exporter/prometheus"
+	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sirupsen/logrus"
 	"go.opencensus.io/stats/view"
-	"open-match.dev/open-match/internal/config"
 )
 
 const (
 	// ConfigNameEnableMetrics indicates that telemetry is enabled.
 	ConfigNameEnableMetrics = "telemetry.prometheus.enable"
 )
 
-func bindPrometheus(mux *http.ServeMux, cfg config.View) {
+func bindPrometheus(p Params, b Bindings) error {
+	cfg := p.Config()
+
 	if !cfg.GetBool("telemetry.prometheus.enable") {
 		logger.Info("Prometheus Metrics: Disabled")
-		return
+		return nil
 	}
 
 	endpoint := cfg.GetString("telemetry.prometheus.endpoint")
+
+	logger.WithFields(logrus.Fields{
+		"endpoint": endpoint,
+	}).Info("Prometheus Metrics: ENABLED")
+
 	registry := prometheus.NewRegistry()
 	// Register standard prometheus instrumentation.
-	registry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
-	registry.MustRegister(prometheus.NewGoCollector())
+	err := registry.Register(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
```

Comment:
I'm not changing that in this PR?  I only changing how errors are handled (returned vs panic).

Reply:
Documentation says do avoid DefaultRegisterer if you want to avoid global state (which I would prefer) https://pkg.go.dev/github.com/prometheus/client_golang/prometheus?tab=doc#pkg-variables